crates/net/eth-wire/src/multiplex.rs

```diff
@@ -166,7 +166,10 @@ impl<St> RlpxProtocolMultiplexer<St> {
             tokio::select! {
                 Some(Ok(msg)) = self.inner.conn.next() => {
                     // Ensure the message belongs to the primary protocol
-                    let offset = msg[0];
+                    let offset = match msg.get(0) {
+                        Some(&byte) => byte,
+                        None => return Err(P2PStreamError::EmptyProtocolMessage.into())
+                    };
```

Comment:
style: this can be `let Some (offset) msg.first() else return` `

Reply:
> style: this can be `let Some (offset) = msg.first() else return` `

The return Err is right though, right?